js/client-teambuilder.js

```diff
@@ -1330,6 +1330,24 @@
 			this.curChartName = '';
 			this.update();
 			this.$('input[name=pokemon]').select();
+			if (this.curTeam.format.includes('monotype')) {
+				var typeTable = [];
+				for (var i = 0; i < this.curSetList.length; i++) {
+					var species = Dex.forGen(this.curTeam.gen).species.get(this.curSetList[i].species);
+					if (!species.exists) continue;
+					if (i === 0) {
+						typeTable = species.types;
+					} else {
+						typeTable = typeTable.filter(function (type) {
+							return species.types.includes(type);
+						});
+					}
+				}
+				if (typeTable.length !== 1) return;
+				this.search.engine.addFilter(['type', typeTable[0]]);
+				this.search.filters = this.search.engine.filters;
+				this.search.find('');
```

Comment:
```suggestion
				if (typeTable.length === 1) {
					this.search.engine.addFilter(['type', typeTable[0]]);
					this.search.filters = this.search.engine.filters;
					this.search.find('');
				}
```

Early return can make things nicer but I think this is the wrong place for it.